src/OpenSSL/SSL.py

```diff
@@ -161,6 +164,47 @@ class SysCallError(Error):
     pass
 
 
+def _make_buffer_checker(cffi_version_string):
+    """
+    Returns a callable that converts a sendable buffer object into something
+    that can be sent by OpenSSL using CFFI. Has different supported types
+    depending on the CFFI version.
+
+    :param cffi_version_string: The version of CFFI in use, from
+        ffi.__version__.
+    """
+    cffi_version = parse(cffi_version_string)
+    if cffi_version >= Version("1.7"):
```

Comment:
The [docs](https://cffi.readthedocs.io/en/latest/ref.html#ffi-buffer-ffi-from-buffer) say "bytearray objects were supported in version 1.7 onwards ... and byte strings were supported in version 1.8 onwards." So the minimum version should probably be 1.8, not 1.7?